popde/density_estimate.py

```diff
@@ -1,3 +1,83 @@
+class SimpleKernelDensityEstimation:
+    """
+    Kernel Density Estimation (KDE) using Kernels
+
+    Methods:
+    --------
+    fit(data, bandwidth=None, kernel=None):
+        Fit the KDE to new data.
+
+    evaluate(points):
+        Evaluate the KDE at given data points.
+
+    Examples:
+    ---------
+    # Create a KDE instance and fit it to data
+    data = np.random.randn(100)
+    kde = SimpleGaussianKernelDensityEstimation(data)
+    kde.fit()
+
+    # Evaluate the KDE at new data points
+    new_data = np.linspace(-3, 3, 100)
+    density_values = kde.evaluate(new_data)
+    """
+
+    def __init__(self, data, bandwidth=1.0, kernel=['gaussian']):
```

Comment:
Also I'd suggest to allow an optional kwarg which is a list of the dimension names (ie names of variables like mass, redshift, chi_eff) and then if it is given to check that the dimensionality of the data array is equal to the number of dimension names.

Reply:
added a bit of  code. need to check if this make sense to you?